ibis/backends/clickhouse/compiler/relations.py

```diff
@@ -17,79 +16,94 @@ def translate_rel(op: ops.TableNode, **_):
     raise com.OperationNotDefinedError(f"No translation rule for {type(op)}")
 
 
-@translate_rel.register(ops.DummyTable)
-def _dummy(op: ops.DummyTable, **kw):
-    return sg.select(
-        *map(partial(translate_val, **kw), op.values), dialect="clickhouse"
-    )
+@translate_rel.register
+def _dummy(op: ops.DummyTable, *, values, **_):
+    return sg.select(*values)
 
 
-@translate_rel.register(ops.PhysicalTable)
+@translate_rel.register
 def _physical_table(op: ops.PhysicalTable, **_):
-    return sg.parse_one(op.name, into=sg.exp.Table)
+    return sg.table(op.name)
+
+
+@translate_rel.register
+def _database_table(op: ops.DatabaseTable, *, name, namespace, **_):
+    return sg.table(name, db=namespace)
+
 
+def replace_tables_with_star_selection(node):
+    if isinstance(node, (sg.exp.Subquery, sg.exp.Table, sg.exp.CTE)):
+        return sg.exp.Column(this=STAR, table=sg.to_identifier(node.alias_or_name))
+    return node
 
-@translate_rel.register(ops.Selection)
-def _selection(op: ops.Selection, *, table, needs_alias=False, **kw):
+
+@translate_rel.register
+def _selection(
+    op: ops.Selection, *, table, selections, predicates, sort_keys, aliases, **_
+):
     # needs_alias should never be true here in explicitly, but it may get
     # passed via a (recursive) call to translate_val
-    assert not needs_alias, "needs_alias is True"
-    if needs_alias := isinstance(op.table, ops.Join) and not isinstance(
+    if isinstance(op.table, ops.Join) and not isinstance(
         op.table, (ops.LeftSemiJoin, ops.LeftAntiJoin)
     ):
         args = table.this.args
         from_ = args["from"]
         (join,) = args["joins"]
     else:
         from_ = join = None
-    tr_val = partial(translate_val, needs_alias=needs_alias, **kw)
-    selections = tuple(map(tr_val, op.selections)) or "*"
-    sel = sg.select(*selections, dialect="clickhouse").from_(
-        from_ if from_ is not None else table, dialect="clickhouse"
-    )
+
+    selections = tuple(
+        replace_tables_with_star_selection(node) for node in selections
+    ) or (STAR,)
+
+    # if we're only reprojecting table, return the table
+    if (
+        len(selections) == 1
+        and from_ is None
+        and join is None
+        and not predicates
+        and not sort_keys
+        and all(
+            isinstance(node, sg.exp.Column) and node.this == STAR for node in selections
+        )
```

Comment:
Not a huge fan of this, but without it, queries like `table.*` reference the wrong alias.

Will investigate further.

Reply:
Only a single test fails when this code is deleted: `ibis/backends/clickhouse/tests/test_select.py::test_filter_predicates`

Reply:
The problem is that we need to replace the `.*` table with the one we're about to select from, in every case **except** selection from a join, because joins don't have a single alias to select from.

I found a slightly better solution.